modules/oe_corporate_countries_address/src/EventSubscriber/AvailableCountriesSubscriber.php

```diff
@@ -52,10 +52,12 @@ public function removeDeprecatedCountries(AvailableCountriesEvent $event) {
     // If no available countries are passed, default it to all available ones.
     if (empty($available_countries)) {
       $available_countries = array_column($this->corporateCountryRepository->getCountries(), 'alpha-2');
+      $available_countries = array_combine($available_countries, $available_countries);
     }
 
     // Extract the alpha-2 of all deprecated countries.
     $deprecated_countries = array_column($this->corporateCountryRepository->getDeprecatedCountries(), 'alpha-2');
+    $deprecated_countries = array_combine($deprecated_countries, $deprecated_countries);
```

Comment:
But if I remove this line, the test still passes. 
So, in order to make sure deprecated countries are removed and this line is actually fixing the problem, the test needs to be improved in order to fail without the fix.

Reply:
To do the test we could do:
```
    // Test: Check explicitly that the actual array does not contain a deprecated country.
    $this->assertArrayNotHasKey('AN', $actual_countries, 'The deprecated country "AN" is not present in actual_countries array');
```

But in reality it is not needed, so I just will keep it here as a comment.

Reply:
Please let me know what you think. Thanks.